src/KapetaAPI.ts

```diff
@@ -401,62 +400,48 @@ export class KapetaAPI {
             headers: {
                 'content-type': 'application/x-www-form-urlencoded',
                 accept: 'application/json',
-            },
+            } as any,
             method: 'POST',
             body: new URLSearchParams({
                 ...payload,
                 client_id: this.getClientId(),
-            }).toString(),
+            }) as any,
         });
     }
 
-    private async _send(opts: Options & { url: string }, retryWithAuth: boolean = true): Promise<any> {
-        return new Promise((resolve, reject) => {
-            request(opts, async (err: any, response: Response, responseBody: any) => {
-                if (err) {
-                    reject(err);
-                    return;
+    private async _send(opts: RequestInit & { url: string }, retryWithAuth: boolean = true): Promise<any> {
+        return fetch(opts.url, opts).then(async (response) => {
+            if (response.status > 299) {
+                if (response.status === 404) {
+                    return null;
                 }
 
-                if (response.statusCode > 299) {
-                    if (response.statusCode === 404) {
-                        resolve(null);
-                        return;
-                    }
-
-                    let errorBody = { error: 'Unknown error', status: response.statusCode };
-                    try {
-                        errorBody = JSON.parse(responseBody);
-                    } catch (e) {}
-
-                    if (retryWithAuth && errorBody.error === 'Token has expired') {
```

Comment:
You're getting rid of this ?

Reply:
If we want to do this - make sure you're not breaking code that depends on it. It should be 1:1 with current functionality

Reply:
No, the fetch handler has this feature too.

Reply:
Not sure I understand? This is checking for specific errors we send internally and reacting to those? I'm sure fetch does not handle that?  :)

Reply:
Ah - now I see it :)